src/hydra-eval-jobs/hydra-eval-jobs.cc

```diff
@@ -504,6 +504,16 @@ int main(int argc, char * * argv)
 
             job.erase("namedConstituents");
 
+            /* Register the derivation as a GC root.  !!! This
+                registers roots for jobs that we may have already
+                done. */
+            auto localStore = store.dynamic_pointer_cast<LocalFSStore>();
+            if (gcRootsDir != "" && localStore) {
+                Path root = gcRootsDir + "/" + std::string(baseNameOf((std::string) job["drvPath"]));
+                if (!pathExists(root))
+                    localStore->addPermRoot(localStore->parseStorePath((std::string) job["drvPath"]), root);
```

Comment:
https://github.com/nlohmann/json#implicit-conversions
this says that we shouldn't use implicit casting for getting json values out of a json object, so likely this should be changed to `job["drvPath"].get<std::string>()`


Maybe something like the following?
```suggestion
                auto drvPath = job["drvPath"].get<std::string>();
                Path root = gcRootsDir + "/" + std::string(baseNameOf(drvPath));
                if (!pathExists(root))
                    localStore->addPermRoot(localStore->parseStorePath(drvPath), root);
```

Reply:
I applied this patch locally and everything looks good!